File: ratemyprofessors/src/main/java/com/tevinjeffrey/rmp/common/Rating.java
package com.tevinjeffrey.rmp.common;

import static com.tevinjeffrey.rmp.common.RMP.RMP_BASE_URL;

public class Rating {

  public double overall;
  public double helpfulness;
  public double clarity;
  public double easiness;
  public int ratingsCount;
  public boolean isHotness;
  public String ratingUrl;
  public String averageGrade;

  @Override
  public String toString() {
    return "Rating{" +
        "overall=" + overall +
        ", helpfulness=" + helpfulness +
        ", clarity=" + clarity +
        ", easiness=" + easiness +
        ", ratingsCount=" + ratingsCount +
        ", isHotness=" + isHotness +
        ", ratingUrl='" + ratingUrl + '\'' +
        ", averageGrade='" + averageGrade + '\'' +
        '}';
  }

  public String getAverageGrade() {
    return averageGrade;
  }

  public double getClarity() {
    return clarity;
  }

  public double getEasiness() {
    return easiness;
  }

  public String getFullRatingUrl() {
    return RMP_BASE_URL + ratingUrl;
  }

  public double getHelpfulness() {
    return helpfulness;
  }

  public double getOverall() {
    return overall;
  }

  public String getRatingUrl() {
    return ratingUrl;
  }

  public int getRatingsCount() {
    return ratingsCount;
  }

  public boolean isHotness() {
    return isHotness;
  }
}

File: ratemyprofessors/src/main/java/com/tevinjeffrey/rmp/common/Rating.kt
package com.tevinjeffrey.rmp.common

data class Rating(
    var overall: Double = 0.toDouble(),
    var helpfulness: Double = 0.toDouble(),
    var clarity: Double = 0.toDouble(),
    var easiness: Double = 0.toDouble(),
    var ratingsCount: Int = 0,
    var isHotness: Boolean = false,
    var ratingUrl: String? = null,
    var averageGrade: String? = null) {

  val fullRatingUrl: String
    get() = RMP.RMP_BASE_URL + ratingUrl!!
}
